clis/teliod/src/qnap.rs

```diff
@@ -0,0 +1,431 @@
+        futures::executor::block_on(tokio::time::timeout(
+            std::time::Duration::from_secs(TIMEOUT_SEC),
+            DaemonSocket::send_command(
+                &socket_path.unwrap_or_default(),
+                &serde_json::to_string(&$command).unwrap_or_default(),
+            ),
+        ))
+    }};
+}
+
+pub(crate) fn handle_request(request: Request) -> Response {
+    match (request.method(), request.uri().query()) {
+        (&Method::POST, _) => start_daemon(),
+        (&Method::DELETE, _) => stop_daemon(),
+        (&Method::PATCH, _) => {
+            let body = match str::from_utf8(request.body()) {
+                Ok(body) => body,
+                Err(error) => {
+                    return text_response(
+                        StatusCode::BAD_REQUEST,
+                        format!("Invalid UTF-8 in request body: {}", error),
+                    )
+                }
+            };
+            update_config(body)
+        }
+        (&Method::GET, Some("info=get-status")) => get_status(),
+        (&Method::GET, Some("info=get-teliod-logs")) => get_teliod_logs(),
+        (&Method::GET, Some("info=get-meshnet-logs")) => get_meshnet_logs(),
+        (_, _) => text_response(StatusCode::BAD_REQUEST, "Invalid request."),
+    }
+}
+
+fn is_teliod_running() -> bool {
+    matches!(teliod_blocking_query!(ClientCmd::IsAlive), Ok(Ok(_)))
+}
+
+fn shutdown_teliod() -> Result<(), TeliodError> {
+    if let Ok(Ok(daemon_reply)) = teliod_blocking_query!(ClientCmd::QuitDaemon) {
+        if CommandResponse::deserialize(&daemon_reply)
+            .is_ok_and(|response| response == CommandResponse::Ok)
+        {
+            return Ok(());
+        }
+    }
+    Err(TeliodError::ClientTimeoutError)
+}
+
+fn start_daemon() -> Response {
+    if is_teliod_running() {
+        return text_response(StatusCode::BAD_REQUEST, "Application is already running.");
+    }
+
+    let mut teliod_log_file = match fs::OpenOptions::new()
+        .create(true)
+        .write(true)
+        .truncate(true)
+        .open(TELIOD_LOG)
+    {
+        Ok(file) => file,
+        Err(_) => {
+            return text_response(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                "Failed to open teliod log file.",
+            );
+        }
+    };
+    let stdout = match teliod_log_file.try_clone() {
+        Ok(file) => Stdio::from(file),
+        Err(error) => {
+            return text_response(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                format!("Failed to start the application: {error}"),
+            )
+        }
+    };
+    let stderr = match teliod_log_file.try_clone() {
+        Ok(file) => Stdio::from(file),
+        Err(error) => {
+            return text_response(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                format!("Failed to start the application: {error}"),
+            )
+        }
+    };
+    match Command::new("setsid")
```

Comment:
imho it would be nicer to fork than using cli

Reply:
`fork()` inherits the parent session id which makes the process quit after the parent session ends. Instead `setsid` creates a new session id for the child process.

https://man7.org/linux/man-pages/man2/setsid.2.html

Reply:
I meant sth like we had in Tcli, it worked well

Reply:
Can you be more specific pls?

Reply:
To be continued on a next PR